addons/editor-messages/userscript.js

```diff
@@ -0,0 +1,60 @@
+export default async function ({ addon, global, console }) {
+  while (true) {
+    const nav = await addon.tab.waitForElement("[class^='menu-bar_account-info-group'] [class^='menu-bar_menu-bar-item']", { markAsSeen: true });
+    const messages = document.createElement("a")
+    messages.href = "/messages/"
+    messages.title = "Messages"
+    messages.setAttribute("style", `
+    background-image: url(/images/nav-notifications.png);
+    background-repeat: no-repeat;
+    background-position: center center;
+    background-size: 45%;
+    padding-right: 10px;
+    padding-left: 10px;
+    width: 30px;
+    overflow: hidden;
+    text-indent: 50px;
+    white-space: nowrap;
+    display: block;
+    padding: 13px 15px 4px 15px;
+    height: 33px;
+    text-decoration: none;
+    color: white;
+    font-size: .85rem;
+    font-weight: bold;
+    `)
+    messages.classList.add("sa-editormessages")
+    let messageCount = document.createElement("span")
+    messages.appendChild(messageCount)
+    if (!nav.querySelector("img")) {
+      nav.appendChild(messages)
+      const setMessages = async () => {
+        fetch(`https://api.scratch.mit.edu/users/${addon.auth.username}/messages/count/`)

```

Comment:
Use `addon.account.getMsgCount()`

Reply:
It was doing a ton of `console.log`s
Also, anything after that code doesn't run:
```javascript
const setMessages = async () => {
        const msgCount = await addon.account.getMsgCount()
        console.log(msgCount);
      };
setMessages();
setInterval(setMessages, 5000);
```
The console.log here doesn't run...

Reply:
Haven't we concluded `console.log` works weirdly on editor pages? Can you try something else to test it?

Reply:
I tried:
```javascript
const setMessages = async () => {
        // fetch(`https://api.scratch.mit.edu/users/${addon.auth.username}/messages/count/`)
        const msgCount = await addon.account.getMsgCount()
        messageCount.innerText = data.count;
        if (msgCount == 0) {
          messageCount.setAttribute("style", `display: none;`);
        } else {
          messageCount.setAttribute("style", "");
        }
      };
      setMessages();
      setInterval(setMessages, 5000);
```
And the innerText of the messageCount was still left empty...

Reply:
Why are you printing `data.count`?

Reply:
Didn't make a difference...
I can literally add anything after it, and it doesn't produce a single error...

I think everything is wrong with the editor... :(

Reply:
I mean, I do not understand why you wrote `messageCount.innerText = data.count;`. What's `data.count`?? Did you mean `messageCount`?

Reply:
Yeah. that was what I meant, I had it in the code, but it still did not do anything.